shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/ManagedBridgeServiceImpl.java

```diff
@@ -1,35 +1,184 @@
+                .withName(managedBridge.getMetadata().getName())
+                .get();
+    }
+
+    @Override
+    public ConfigMap fetchOrCreateBridgeConfigMap(ManagedBridge managedBridge, Secret secret) {
+        ConfigMap expected = templateProvider.loadBridgeIngressConfigMapTemplate(managedBridge, TemplateImportConfig.withDefaults(LabelsBuilder.V2_OPERATOR_NAME));
+
+        expected.getData().replace(GlobalConfigurationsConstants.KNATIVE_KAFKA_TOPIC_PARTITIONS_CONFIGMAP, GlobalConfigurationsConstants.KNATIVE_KAFKA_TOPIC_PARTITIONS_VALUE_CONFIGMAP); // TODO: move to DTO?
+        expected.getData().replace(GlobalConfigurationsConstants.KNATIVE_KAFKA_REPLICATION_FACTOR_CONFIGMAP, GlobalConfigurationsConstants.KNATIVE_KAFKA_REPLICATION_FACTOR_VALUE_CONFIGMAP); // TODO: move to DTO?
+        expected.getData().replace(GlobalConfigurationsConstants.KNATIVE_KAFKA_TOPIC_BOOTSTRAP_SERVERS_CONFIGMAP,
+                new String(Base64.getDecoder().decode(secret.getData().get(GlobalConfigurationsConstants.KNATIVE_KAFKA_BOOTSTRAP_SERVERS_SECRET))));
+        expected.getData().replace(GlobalConfigurationsConstants.KNATIVE_KAFKA_TOPIC_SECRET_REF_NAME_CONFIGMAP, secret.getMetadata().getName());
+        expected.getData().replace(GlobalConfigurationsConstants.KNATIVE_KAFKA_TOPIC_TOPIC_NAME_CONFIGMAP,
+                new String(Base64.getDecoder().decode(secret.getData().get(GlobalConfigurationsConstants.KNATIVE_KAFKA_TOPIC_NAME_SECRET))));
+
+        ConfigMap existing = kubernetesClient
+                .configMaps()
+                .inNamespace(managedBridge.getMetadata().getNamespace())
+                .withName(managedBridge.getMetadata().getName())
+                .get();
+
+        if (existing == null || !expected.getData().equals(existing.getData())) {
+            return kubernetesClient
+                    .configMaps()
+                    .inNamespace(managedBridge.getMetadata().getNamespace())
+                    // Best practice would be to generate a new name for the configmap and replace its reference
+                    .withName(managedBridge.getMetadata().getName())
+                    .createOrReplace(expected);
+        }
+
+        return existing;
+    }
+
+    @Override
+    public KnativeBroker fetchOrCreateKnativeBroker(ManagedBridge managedBridge, ConfigMap configMap) {
+
+        KnativeBroker expected = templateProvider.loadBridgeIngressBrokerTemplate(managedBridge, TemplateImportConfig.withDefaults(LabelsBuilder.V2_OPERATOR_NAME));
+        expected.getSpec().getConfig().setName(configMap.getMetadata().getName());
+        expected.getSpec().getConfig().setNamespace(configMap.getMetadata().getNamespace());
+        expected.getMetadata().getAnnotations().replace(GlobalConfigurationsConstants.KNATIVE_BROKER_EXTERNAL_TOPIC_ANNOTATION_NAME,
+                configMap.getData().get(GlobalConfigurationsConstants.KNATIVE_KAFKA_TOPIC_TOPIC_NAME_CONFIGMAP));
```

Comment:
first we fetch topicName from BridgeDTO and put it inside secret
then take it from secret and put it inside configmap
later fetch it from configmap and put it inside Knative
```
BridgeDTO -> Secret
Secret -> ConfigMap
ConfigMap -> KnativeBroker
```

Why we are not setting `TopicName` inside `ManagedBridge` CRD as discussed in MGDOBR-1186 approach?

Reply:
@vaibhavjainwiz The topic name is set on the `ManagedBridge` CRD. We're going to be constrained by what Knative eventing requires us to do and it requires that the topic name and other Kafka configuration is provided by a `Secret`.

As I've mentioned, please consider this PR a "get it all working" first pass. Once it all works, we can start to make smaller refactors where it makes sense to do so.